autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -66,6 +75,31 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
             clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
             LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
                     cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+        } else if (RUNNING.equals(cluster.getState()) && (cluster.getMachineUserCrn() == null || cluster.getEnvironmentCrn() == null)) {
+            populateEnvironmentAndMachineUserIfNotPresent(cluster);
+        }
+    }
+
+    protected void beforeDeleteCleanup(Cluster cluster) {
+        try {
+            if ((cluster.getEnvironmentCrn() != null) && (clusterService.countByEnvironmentCrn(cluster.getEnvironmentCrn()) <= 1)) {
+                altusMachineUserService.deleteMachineUserForEnvironment(cluster.getClusterPertain().getTenant(),
+                        cluster.getMachineUserCrn(), cluster.getEnvironmentCrn());
+            }
+        } catch (Exception ex) {
+            LOGGER.warn("Error deleting machineUserCrn '{}' for environment '{}'",
+                    cluster.getMachineUserCrn(), cluster.getEnvironmentCrn(), ex);
+        }
+    }
+
+    protected void populateEnvironmentAndMachineUserIfNotPresent(Cluster cluster) {
```

Comment:
Does this need to be applied in the new cluster deletion handler as well?

Reply:
Added the `beforeDeleteCleanup` method present above this one to the delete handler, since we require the machineUser to be deleted for terminated clusters.